### PR TITLE
Some more builtins

### LINKED ISSUE #96: Elpi predicate to force garbage collection
I wish to have an Elpi predicate to force a garbage collection. There are some OCaml functions (`Gc.minor`, `Gc.major`, `Gc.full_major`, and `Gc.compact`) to do that and `Gc.compact` is most desirable among those for me. @gares

### PR DESCRIPTION
**Pull request by gares**
Fix #96

### CODE REVIEW

File: CHANGES.md [hunk 1]
```diff
@@ -1,3 +1,19 @@
+# UNRELEASED
+
+- API:
+  - New `gc.control` type for the garbage collector settings
+  - New `gc.get` and `gc.set` for reading and writing GC settings
+  - New `gc.minor`
+  - New `gc.major`
+  - New `gc.full`
```
Comment (pi8027) [inline CHANGES.md:8]: `gc.compact` is missing here.

Comment (gares) [reply CHANGES.md:8]: Thanks for spotting (it is in the new API indeed)

**Comment by pi8027** [comment]
Should they also be exported from `elpi-builtin.elpi` of Coq-Elpi if I wish to use them from Coq-Elpi? It says `constant gc.compact has no declared type` and then `elpi fails`.

**Comment by gares** [comment]
Well spotted. My last commit exports a new list of builtin declarations, which should be passed to API.Setup.init in coq_elpi_vernacular.ml

**Review by pi8027 — APPROVED** [review]
By inserting `gc.compact`, I could reduce memory usage. So it seems to work!

**Comment by pi8027** [comment]
Now I think `Gc.stat` and `Gc.quick_stat` are useful to check memory consumption of some computation. Indeed, it is not impossible (intentional double negation ;)) to do so using `Gc.print_stat` but would be easier with `Gc.stat`. Would you mind adding them? (Sorry for asking so many feature requests.)

**Comment by gares** [comment]
What about removing print_stat and keeping only stat and quick_stat ?

**Comment by pi8027** [comment]
> What about removing print_stat and keeping only stat and quick_stat ?

Totally fine with me.

**Comment by gares** [comment]
I've updated the API

**Comment by pi8027** [comment]
I couldn't build Coq-Elpi with this patch and so cannot test:
```
### output ###
# [...]
# CAMLC -c src/coq_elpi_HOAS.ml
# File "src/coq_elpi_HOAS.ml", line 658, characters 33-112:
# 658 |     deprecation = deprecation @@ get_pair_option API.BuiltInData.string API.BuiltInData.string "coq:deprecation";
#                                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
# Error: This expression has type
#          (string Elpi.Builtin.unspec * string Elpi.Builtin.unspec) option
#        but an expression was expected of type
#          (string unspec * string unspec) option
#        Type string Elpi.Builtin.unspec is not compatible with type
#          string unspec
# make[1]: *** [Makefile.coq:663: src/coq_elpi_HOAS.cmo] Error 2
# make: *** [Makefile:30: build] Error 2
```

**Comment by pi8027** [comment]
I managed to fix and now I can do this in Coq-Elpi:
```
Elpi Query lp:{{ gc.stat Minor Promoted Major _ _ _ _ _ _ _ _ _ _ _ _ _ }}.
Elpi Query lp:{{ gc.get Minor _ _ _ _ _ _ _ }}.
Elpi Query lp:{{ gc.set 67108864 _ _ _ _ _ _ _ }}.
```

**Review by pi8027 — APPROVED** [review]
LGTM

**Comment by gares** [comment]
Oh right, I took the occasion to "move" the unspec data type to elpi, so one has to remove it's copy in coq-elpi

File: src/builtin.mli [hunk 2]
```diff
@@ -54,6 +57,10 @@ val diagnostic : diagnostic API.Conversion.t
 val mkOK : diagnostic
 val mkERROR : string -> diagnostic
 
+type 'a unspec = Given of 'a | Unspec
+val unspec : 'a API.Conversion.t -> 'a unspec API.Conversion.t
+val unspecC : ('a,'b,'c) API.ContextualConversion.t -> ('a unspec,'b,'c) API.ContextualConversion.t
```
Comment (pi8027) [inline src/builtin.mli:62]: I suggest relocating them to `Elpi.API`. (see LPCIC/coq-elpi#230.)

File: src/builtin.elpi [hunk 3]
```diff
@@ -119,6 +119,14 @@ type iabs int -> int.
 
 type rabs float -> float.
 
+type (max) int -> int -> int.
+
+type (max) float -> float -> float.
+
+type (min) int -> int -> int.
+
+type (min) float -> float -> float.
```
Comment (pi8027) [inline src/builtin.elpi:128]: I'm not sure whether they should really be infix.

Comment (gares) [reply src/builtin.elpi:128]: They are not, it's a bug in the pretty printer but has no effect in practice.

**Pull request merged**